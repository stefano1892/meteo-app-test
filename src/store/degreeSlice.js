import { createSlice } from "@reduxjs/toolkit";

const initialState = {
    degree: "Celsius"
}

export const degreeSlice = createSlice({
    name: 'degreeSelection',
    initialState,
    reducers: {
        changeDegree: (state, action) => {
            state.degree = action.payload
        }
    }
})

export const { changeDegree }  = degreeSlice.actions;

export default degreeSlice.reducer;