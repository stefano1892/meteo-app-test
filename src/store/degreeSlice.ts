import { createSlice } from "@reduxjs/toolkit";

interface InitialStateTypes {
    degree: string
}


const initialState: InitialStateTypes = {
    degree: "Celsius"
}

export const degreeSlice = createSlice({
    name: 'degreeSelection',
    initialState,
    reducers: {
        changeDegree: (state, action) => {
            state.degree = action.payload
        }
    }
})

export const { changeDegree }  = degreeSlice.actions;

export default degreeSlice.reducer;