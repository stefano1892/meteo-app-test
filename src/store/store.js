import { configureStore } from '@reduxjs/toolkit'
import weaterReducer from './weatherSlice'
import degreeReducer from './degreeSlice'

export const store = configureStore({
  reducer: {
    pastResearch: weaterReducer,
    degreeSelection: degreeReducer
  },
})