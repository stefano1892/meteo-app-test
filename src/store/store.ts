import { configureStore } from '@reduxjs/toolkit'
import weaterReducer from './weatherSlice'
import degreeReducer from './degreeSlice'

export const store = configureStore({
  reducer: {
    pastResearch: weaterReducer,
    degreeSelection: degreeReducer
  },
})

export type AppSelector = ReturnType<typeof store.getState>

// () => number => ReturnType<() => number> = number