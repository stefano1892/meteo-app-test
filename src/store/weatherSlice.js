import { createSlice } from "@reduxjs/toolkit";

const initialState = {
    searches: []
}

export const weatherSlice = createSlice({
    name: 'pastResearch',
    initialState,
    reducers: {
        addResearch: (state, action) => {
            state.searches.push(action.payload)
            if(state.searches.length > 5) {
                state.searches.shift()
            }
        }
    }
})

export const { addResearch }  = weatherSlice.actions;

export default weatherSlice.reducer;