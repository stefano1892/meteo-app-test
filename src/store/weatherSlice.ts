import { createSlice } from "@reduxjs/toolkit";
import { ICity } from "../interfaces/ICity";

interface InitialStateTypes {
    searches: ICity[]
}

const initialState: InitialStateTypes = {
    searches: []
}

export const weatherSlice = createSlice({
    name: 'pastResearch',
    initialState,
    reducers: {
        addResearch: (state, action) => {
            state.searches.push(action.payload)
            if(state.searches.length > 5) {
                state.searches.shift()
            }
        }
    }
})

export const { addResearch }  = weatherSlice.actions;

export default weatherSlice.reducer;