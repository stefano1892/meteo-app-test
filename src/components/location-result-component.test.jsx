import { Provider } from "react-redux"
import { render } from "@testing-library/react"
import { store } from "../store/store"
import LocationResultComponent from "./location-result-component"

describe('LocationResultComponent', () => {
    test('Component must return nothing if city is not defined', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <LocationResultComponent city={undefined}/>
            </Provider>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('Component must return something if city is defined', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <LocationResultComponent city={{
       "request":{
          "type":"City",
          "query":"Milan, Italy",
          "language":"en",
          "unit":"m"
       },
       "location":{
          "name":"Milan",
          "country":"Italy",
          "region":"Lombardia",
          "lat":"45.467",
          "lon":"9.200",
          "timezone_id":"Europe/Rome",
          "localtime":"2024-01-12 21:10",
          "localtime_epoch":1705093800,
          "utc_offset":"1.0"
       },
       "current":{
          "observation_time":"08:10 PM",
          "temperature":3,
          "weather_code":113,
          "weather_icons":[
             "https://cdn.worldweatheronline.com/images/wsymbols01_png_64/wsymbol_0008_clear_sky_night.png"
          ],
          "weather_descriptions":[
             "Clear"
          ],
          "wind_speed":4,
          "wind_degree":97,
          "wind_dir":"E",
          "pressure":1024,
          "precip":0,
          "humidity":93,
          "cloudcover":0,
          "feelslike":2,
          "uv_index":1,
          "visibility":10,
          "is_day":"no"
       }
    }}/>
            </Provider>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})