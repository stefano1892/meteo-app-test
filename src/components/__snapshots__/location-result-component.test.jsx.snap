// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationResultComponent Component must return nothing if city is not defined 1`] = `<DocumentFragment />`;

exports[`LocationResultComponent Component must return something if city is defined 1`] = `
<DocumentFragment>
  <div
    class="m-0 w-100 row"
  >
    <div
      class="location-header"
    >
      <b>
        Posizione
      </b>
    </div>
  </div>
  <div
    class="m-0 w-100 mt-5 row"
  >
    <div
      class="col"
    >
      <div
        class="location-name"
      >
        Milan
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="text-center"
      >
        <img
          alt=""
          src="https://cdn.worldweatheronline.com/images/wsymbols01_png_64/wsymbol_0008_clear_sky_night.png"
        />
      </div>
    </div>
  </div>
  <div
    class="m-0 w-100 mt-5 row"
  >
    <div
      class="col"
    >
      <div
        class="location-list-container"
      >
        <ul
          class="location-list"
        >
          <li>
            Paese: Italy
          </li>
          <li>
            Regione: Lombardia
          </li>
          <li>
            Latitudine: 45.467
          </li>
          <li>
            Longitudine: 9.200
          </li>
        </ul>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="location-list-container"
      >
        <ul
          class="location-list"
        >
          <li>
            Timezone: Europe/Rome
          </li>
          <li>
            Data e ora: 2024-01-12 21:10
          </li>
          <li>
            Fuso orario: 1.0
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
